pkg/util/selector_test.go

```diff
@@ -0,0 +1,854 @@
+		},
+		{
+			name: "match all",
+			args: args{
+				rs: matchAll,
+			},
+			want: true,
+		},
+		{
+			name: "match labels",
+			args: args{
+				rs: matchLabels,
+			},
+			want: true,
+		},
+		{
+			name: "match name",
+			args: args{
+				rs: matchName,
+			},
+			want: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := ResourceMatches(resource, tt.args.rs); got != tt.want {
+				t.Errorf("ResourceMatches() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestResourceSelectorPriority(t *testing.T) {
+	resource := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "v1",
+			"kind":       "Pod",
+			"metadata": map[string]interface{}{
+				"name":      "test",
+				"namespace": "default",
+				"labels": map[string]interface{}{
+					"foo": "bar",
+				},
+			},
+		},
+	}
+
+	type args struct {
+		rs policyv1alpha1.ResourceSelector
+	}
+	tests := []struct {
+		name string
+		args args
+		want ImplicitPriority
+	}{
+		{
+			name: "apiserver not matched",
+			args: args{
+				rs: policyv1alpha1.ResourceSelector{
+					APIVersion: "unmatched",
+					Kind:       "Pod",
+					Name:       "foo",
+					Namespace:  "default",
+				},
+			},
+			want: PriorityMisMatch,
+		},
+		{
+			name: "apiserver not matched",
+			args: args{
+				rs: policyv1alpha1.ResourceSelector{
+					APIVersion: "v1",
+					Kind:       "Unmatched",
+					Name:       "foo",
+					Namespace:  "default",
+				},
+			},
+			want: PriorityMisMatch,
+		},
+		{
+			name: "namespace not matched",
+			args: args{
+				rs: policyv1alpha1.ResourceSelector{
+					APIVersion: "v1",
+					Kind:       "Pod",
+					Name:       "foo",
+					Namespace:  "Unmatched",
+				},
+			},
+			want: PriorityMisMatch,
+		},
+		{
+			name: "namespace unset and name matched",
+			args: args{
+				rs: policyv1alpha1.ResourceSelector{
+					APIVersion: "v1",
+					Kind:       "Pod",
+					Name:       "test",
+				},
+			},
+			want: PriorityMatchName,
+		},
+		{
+			name: "[case 1] name not matched",
+			args: args{
+				rs: policyv1alpha1.ResourceSelector{
+					APIVersion: "v1",
+					Kind:       "Pod",
+					Name:       "unmatched",
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"foo": "bar",
+						},
+					},
+				},
+			},
+			want: PriorityMisMatch,
+		},
+		{
+			name: "[case 1] name matched, labels are matched and ignore",
+			args: args{
+				rs: policyv1alpha1.ResourceSelector{
+					APIVersion: "v1",
+					Kind:       "Pod",
+					Name:       "test",
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"foo": "unmatched",
+						},
+					},
+				},
+			},
+			want: PriorityMatchName,
+		},
+		{
+			name: "[case 2] name not matched, labels unset",
+			args: args{
+				rs: policyv1alpha1.ResourceSelector{
+					APIVersion: "v1",
+					Kind:       "Pod",
+					Name:       "unmatched",
+				},
+			},
+			want: PriorityMisMatch,
+		},
```

Comment:
Redundant with previous `[case 1] name not matched`.

Reply:
The `case x` means these scenes:
https://github.com/karmada-io/karmada/blob/a795d147673cb8afea34a59839e6e216555fd1b0/pkg/util/selector.go#L42-L47

In case 2, label is empty. While in case 1 it's set.